users/service.js

```diff
@@ -0,0 +1,66 @@
+var response = require('../Objects/response.js');
+var userModels = require('./userModels');
+var Student = userModels.Student;
+var Admin = userModels.Admin;
+
+var createUser = (json) => {
+  if (json.role == 'Student') {
+    // Create new student.
+    var student = new Student({
+      approvalStatus: false,
+      email: json.email,
+      firstName: json.firstName,
+      lastName: json.lastName,
+      points: 0,
+      role: 'Student',
+      studentId: getUid();
```

Comment:
Once you save this document, mongoose will keep this field 0, and will create a separate field called ```_id``` for the generated UID. I.e., this field would never be used as a UID. There might be a way to set up mongoose to use this field for the UID, but I'd recommend removing this field and just using the default ```_id``` field.